flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -861,6 +869,43 @@ else if (transitionState(current, ExecutionState.FAILED, t)) {
 		}
 	}
 
+	private void waitInvokableCancelCompletion() throws java.util.concurrent.ExecutionException, java.util.concurrent.TimeoutException {
+		if (!invokableHasBeenCanceled.get()) {
+			return;
+		}
+
+		// set deadline if taskCancellationTimeout is set to have time for cleanup (before being killing by TaskCancelerWatchDog) (best effort)
+		Optional<Deadline> deadline = Optional
+			.of((long) (taskCancellationTimeout * .9))
+			.filter(t -> t > 0)
+			.map(t -> Deadline.fromNow(Duration.ofMillis((t))));
+
+		if (taskCancellationTimeout > 0 && !deadline.isPresent()) {
+			LOG.debug("Skip waiting for invokable cancellation to have time for cleanup (taskCancellationTimeout is too small)");
+			return;
+		}
+		LOG.debug("Waiting for invokable cancellation (deadline: {})", deadline);
+
+		// prevent interrupts by TaskInterrupter and other canceling threads (best effort)
+		invokable.setShouldInterruptOnCancel(false);
```

Comment:
This change would be perfectly fine without this line right? In that case I would prefer to remove it, I think it would make the code cleaner/easier to understand.

Reply:
Right. Ok, I'll remove it.